File: examples/src/main/java/template/Serving.java
package myengine;

import io.prediction.controller.java.LJavaServing;

import myengine.MyParams.ServingParams;
import myengine.MyData.Query;
import myengine.MyData.PredictedResult;

public class Serving
  extends LJavaServing<ServingParams, Query, PredictedResult> {

  public Serving(ServingParams params) {

  }

  @Override
  public PredictedResult serve(Query query, Iterable<PredictedResult> predictions) {
    return new PredictedResult();
  }

}

File: templates/scala-parallel-recommendation/src/main/scala/Serving.scala
package org.template.recommendation

import io.prediction.controller.LServing
import io.prediction.controller.EmptyServingParams

class Serving
  extends LServing[EmptyServingParams, Query, PredictedResult] {

  override
  def serve(query: Query,
    predictions: Seq[PredictedResult]): PredictedResult = {
    predictions.head
  }
}
